content/blog/deploying-a-django-application-to-aws/index.md

```diff
@@ -0,0 +1,561 @@
+    password=sql_admin_password,
+    instance_class="db.t2.micro",
+    allocated_storage=20,
+    skip_final_snapshot=True,
+    publicly_accessible=True,
+    db_subnet_group_name=app_database_subnetgroup.id,
+    vpc_security_group_ids=[app_security_group.id])
+```
+
+Pulumi offers some additional tools to make handling MySQL easier.
+
+```python
+mysql_provider = mysql.Provider("mysql-provider",
+    endpoint=mysql_rds_server.endpoint,
+    username=sql_admin_name,
+    password=sql_admin_password)
+```
+
+We initialize the example database and create a user to manage it.
+
+```python
+mysql_database = mysql.Database("mysql-database",
+    name="votes-database",
+    opts=pulumi.ResourceOptions(provider=mysql_provider))
+
+mysql_user = mysql.User("mysql-standard-user",
+    user=sql_user_name,
+    host="example.com",
+    plaintext_password=sql_user_password,
+    opts=pulumi.ResourceOptions(provider=mysql_provider))
+```
+
+The user account which we will give to Django only needs the "SELECT", "UPDATE", "INSERT", 
+and "DELETE" permissions to function.
+
+```python
+mysql_access_grant = mysql.Grant("mysql-access-grant",
+    user=mysql_user.user,
+    host=mysql_user.host,
+    database=mysql_database.name,
+    privileges= ["SELECT", "UPDATE", "INSERT", "DELETE"],
+    opts=pulumi.ResourceOptions(provider=mysql_provider))
+```
+
+Our RDS instance has been set up, and the MySQL backend is complete. All that's left is to 
+create the Django frontend.
+
+A target group, balancer, and listener is created for the frontend.
+
+```python
+django_targetgroup = aws.lb.TargetGroup("django-targetgroup",
+	port=80,
+	protocol="TCP",
+	target_type="ip",
+    stickiness= {
+        "enabled": False,
+        "type": "lb_cookie",
+    },
+	vpc_id=app_vpc.id)
+
+django_balancer = aws.lb.LoadBalancer("django-balancer",
+    load_balancer_type="network",
+    internal=False,
+    security_groups=[],
+    subnets=[app_vpc_subnet.id])
+
+django_listener = aws.lb.Listener("django-listener",
+	load_balancer_arn=django_balancer.arn,
+	port=80,
+    protocol="TCP",
+	default_actions=[{
+		"type": "forward",
+		"target_group_arn": django_targetgroup.arn
+	}])
+```
+
+The application is built into a Docker image and pushed to our ECR repository we created
+earlier.
+
+```python
+def get_registry_info(rid):
+    creds = aws.ecr.get_credentials(registry_id=rid)
+    decoded = base64.b64decode(creds.authorization_token).decode()
+    parts = decoded.split(':')
+    if len(parts) != 2:
+        raise Exception("Invalid credentials")
+    return docker.ImageRegistry(creds.proxy_endpoint, parts[0], parts[1])
+
+app_registry = app_ecr_repo.registry_id.apply(get_registry_info)
+
+django_image = docker.Image("django-dockerimage",
+    image_name=app_ecr_repo.repository_url,
+    build="./frontend",
+    skip_push=False,
+    registry=app_registry
+)
+```
+
+To help with debugging our application, we will create a Cloudwatch instance to automatically 
+store all logs.
```

Comment:
```suggestion
```

Reply:
Would it be better to have "To help with debugging our application, we will create a Cloudwatch instance to store all logs" instead?

Reply:
yes